model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -335,6 +335,12 @@ class DependencyGraphBuilder<D>(
         val dependencies2 = dependencies.associateBy { dependencyHandler.identifierFor(it) }
         if (!dependencies2.keys.containsAll(dependencies1)) return false
 
+        if (dependencyHandler::class.simpleName == "GradleDependencyHandler") {
+            // In case of Gradle, we can skip the costly deep comparison.
+            // If the dependencies are the same, their children are also the same.
+            if (dependencies2.keys.sorted() == dependencies1.sorted()) return true
```

Comment:
So save sorting at least on one side, and as `keys` already is a set, how about

    if (dependencies2.keys == dependencies1.toSet()) return true

instead?